### PR TITLE
fix: Use IPv6 CIDR block destination on route when IPv6 support is enabled

### LINKED ISSUE #101: Using IPv6 with `vpc_attachments[].vpc_route_table_ids` fails
## Description

The module generally supports the use of IPv6. But when adding routes to the tables provided via `vpc_attachments[].vpc_route_table_ids`, it always tries to assign the `vpc_attachments[].tgw_destination_cidr` to the IPv4 `aws_route.destination_cidr_block` argument. Correct would be using `aws_route.destination_cidr_block` for IPv4 and `aws_route.destination_ipv6_cidr_block` for IPv6.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 2.8.2
- Terraform version:
  ```
  Terraform v1.3.7
  on darwin_arm64
  ```
- Provider version(s): 
  ```
  provider registry.terraform.io/hashicorp/aws v4.46.0
  ```

## Reproduction Code [Required]

Everything it needs is to configure IPv6 and provide VPC Route Tables:

```hcl
module "my_vpc_ipv6" {
  source  = "terraform-aws-modules/vpc/aws"
  version = "~> 3.0"

  name = "my-vpc-ipv6"
  cidr = "10.1.0.0/16"

  azs             = ["${local.region}a", "${local.region}b"]
  private_subnets = ["10.1.1.0/24", "10.1.2.0/24"]
  public_subnets  = ["10.1.101.0/24", "10.1.102.0/24"]

  enable_nat_gateway = false

  enable_ipv6                     = true
  assign_ipv6_address_on_creation = true

  private_subnet_assign_ipv6_address_on_creation = false

  public_subnet_ipv6_prefixes  = [0, 1]
  private_subnet_ipv6_prefixes = [2, 3]
}

module "my_tgw_ipv6" {
  source  = "terraform-aws-modules/transit-gateway/aws"
  version = "~> 2.0"

  name = "my-tgw-ipv6"

  vpc_attachments = {
    vpc = {
      vpc_id     = module.my_vpc_ipv6.vpc_id
      subnet_ids = module.my_vpc_ipv6.private_subnets

      ipv6_support         = true
      tgw_destination_cidr = "::/0"
      vpc_route_table_ids  = module.my_vpc_ipv6.private_route_table_ids
    },
  }
}
```

## Expected behavior

The code runs and IPv6 routes are being created.

## Actual behavior

The code would fails because of issue https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/issues/100, but else it would fail, trying to assign an IPv6 CIDR to the IPv4 `aws_route.destination_cidr_block` instead of using its IPv6 counterpart `aws_route.destination_ipv6_cidr_block`.

### LINKED ISSUE #112: Using IPv6 with `vpc_attachments[].vpc_route_table_ids` fails
## Description

ℹ️ This is a copy of #101 which was closed automatically while still valid.

The module generally supports the use of IPv6. But when adding routes to the tables provided via `vpc_attachments[].vpc_route_table_ids`, it always tries to assign the `vpc_attachments[].tgw_destination_cidr` to the IPv4 `aws_route.destination_cidr_block` argument. Correct would be using `aws_route.destination_cidr_block` for IPv4 and `aws_route.destination_ipv6_cidr_block` for IPv6.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 2.8.2
- Terraform version:
  ```
  Terraform v1.3.7
  on darwin_arm64
  ```
- Provider version(s): 
  ```
  provider registry.terraform.io/hashicorp/aws v4.46.0
  ```

## Reproduction Code [Required]

Everything it needs is to configure IPv6 and provide VPC Route Tables:

```hcl
module "my_vpc_ipv6" {
  source  = "terraform-aws-modules/vpc/aws"
  version = "~> 3.0"

  name = "my-vpc-ipv6"
  cidr = "10.1.0.0/16"

  azs             = ["${local.region}a", "${local.region}b"]
  private_subnets = ["10.1.1.0/24", "10.1.2.0/24"]
  public_subnets  = ["10.1.101.0/24", "10.1.102.0/24"]

  enable_nat_gateway = false

  enable_ipv6                     = true
  assign_ipv6_address_on_creation = true

  private_subnet_assign_ipv6_address_on_creation = false

  public_subnet_ipv6_prefixes  = [0, 1]
  private_subnet_ipv6_prefixes = [2, 3]
}

module "my_tgw_ipv6" {
  source  = "terraform-aws-modules/transit-gateway/aws"
  version = "~> 2.0"

  name = "my-tgw-ipv6"

  vpc_attachments = {
    vpc = {
      vpc_id     = module.my_vpc_ipv6.vpc_id
      subnet_ids = module.my_vpc_ipv6.private_subnets

      ipv6_support         = true
      tgw_destination_cidr = "::/0"
      vpc_route_table_ids  = module.my_vpc_ipv6.private_route_table_ids
    },
  }
}
```

## Expected behavior

The code runs and IPv6 routes are being created.

## Actual behavior

The code would fails because of issue https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/issues/100, but else it would fail, trying to assign an IPv6 CIDR to the IPv4 `aws_route.destination_cidr_block` instead of using its IPv6 counterpart `aws_route.destination_ipv6_cidr_block`.

### PR DESCRIPTION
**Pull request by Jarodiv**
## Description
Make route creation use `aws_route.destination_ipv6_cidr_block` for attachments with `ipv6_support` set to `true`.

## Motivation and Context
Creating routes for TGW attachments breaks the generally working support for IPv6 based setups.
- Resolves #101 
- Resolves #112

## Breaking Changes
There should be no negative effect on backwards compatibility with the current major version.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by Jarodiv** [comment]
"Keep alive" post

**Comment by Jarodiv** [comment]
"Keep alive" post

**Comment by Jarodiv** [comment]
"Keep alive" post

**Comment by Jarodiv** [comment]
"Keep alive" post"Keep alive" post

**Comment by Jarodiv** [comment]
"Keep alive" post

**Comment by Jarodiv** [comment]
"Keep alive" post

**Comment by Jarodiv** [comment]
"Keep alive" post

File: main.tf [hunk 1]
```diff
@@ -116,9 +116,10 @@ resource "aws_route" "this" {
     tgw_id = x.tgw_id
   } }
 
-  route_table_id         = each.key
-  destination_cidr_block = each.value["cidr"]
-  transit_gateway_id     = each.value["tgw_id"]
+  route_table_id              = each.key
+  destination_cidr_block      = try(each.value.ipv6_support, false) ? null : each.value["cidr"]
+  destination_ipv6_cidr_block = try(each.value.ipv6_support, false) ? each.value["cidr"] : null
+  transit_gateway_id          = aws_ec2_transit_gateway.this[0].id
```
Comment (bryantbiggs) [inline main.tf:122]:
```suggestion
  transit_gateway_id          = each.value["tgw_id"]
```

**Comment by antonbabenko** [comment]
This PR is included in version 2.12.1 :tada:

**Pull request merged**